main_test.py

```diff
@@ -208,6 +208,21 @@ def test_api_unauthorized(self):
             response = self.client.get(endpoint)
             self.assertEqual(response.status_code, 403)
 
+    # Test that username being set from cookies is what we want
+    def test_login(self):
+        self.client.set_cookie('localhost', 'token', '{"email": "aaardvark@eastsideprep.org"}')
+        test_username = "aaardvark"
+        with self.client as c:
+            response = c.get('/')
+            # Get username from session and compare it to actual one
+            with c.session_transaction() as sess:
+                self.assertEqual(sess["username"], test_username)
+                # Return it to a not logged in state for other tests
+                del sess["username"]

```

Comment:
Rather than having to "fix" the state of the class after each test, best practice would be to delete all cookies in a `setUp` method.

Reply:
or perhaps rely on teardown_class from https://docs.pytest.org/en/latest/xunit_setup.html#method-and-function-level-setup-teardown.

Also, sess.clear() is the preferred way to do this, apparently

Reply:
So can I just do the sess.clear() or should I put it in one of the setup methods?

Reply:
you could put it in the existing setUpClass, but I'd suggest putting it in a new teardown_class method (see link), whose purpose is test cleanup like this.